modules/concurrency/src/spinner.cpp

```diff
@@ -0,0 +1,56 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include "hephaestus/concurrency/spinner.h"
+
+#include <fmt/format.h>
+
+#include "hephaestus/base/exception.h"
+
+namespace heph::concurrency {
+Spinner::Spinner() : is_started_(false) {
+}
+
+Spinner::~Spinner() {
+  try {
+    if (is_started_.load() && spinner_thread_.joinable()) {
+      stop();
+    }
+  } catch (const std::exception& ex) {
+    fmt::print("Error stopping spinner: {}", ex.what());
+  }
```

Comment:
I am wondering whether here we should report an error if stop is not called before the destructor.
Maybe even throw, even if it is not recommended inside a constructor.

Do you remember the start/stop behavior in Arrival?

Reply:
this code is actualy to not throw in the destructor, as it won't be  propagated correctly. Why do you think its not good design here?

Reply:
It should be an error if the user doesn't call stop before the spinner terminates. So I think we should be destructive (pun intended).
Maybe I would even do `std::terminate`.